aten/src/ATen/core/boxing/impl/boxing.h

```diff
@@ -296,6 +300,46 @@ struct BoxedKernelWrapper<
   }
 };
 
+//
+// 4.5. Again, in-progress migration to take and return const Tensor
+// references instead.
+// Note: all signatures matching this pattern are are assumed to be for such ops.
+// This assumption permits the generated BoxedKernelWrapper specializations to simply
+// return out arguments.
```

Comment:
oh this is so cute, you're using the fact that the return type is a `const Tensor&` to infer that the first argument is what we previously treated as a mutable tensor ref (otherwise it is indistinguishable)

cc @bhosmer